src/Hedgehog/paket.template

```diff
@@ -20,3 +20,6 @@ iconUrl
     https://github.com/hedgehogqa/fsharp-hedgehog/raw/master/img/hedgehog-logo.png
 tags
     fsharp, f#, testing
+files
+    *.fsproj ==> fable/
+    **/*.fs ==> fable/
```

Comment:
How does this work? :smiley: I apologize if I'm missing something obvious. I'm just trying to understand how we're going to invoke the Fable compiler as part of the build process.

Reply:
We are not calling fable compiler in the build process we are just adding the source code to the nuget package and later when somebody compiles a fable project with such a nuget package everything works. See here for more https://fable.io/docs/your-fable-project/author-a-fable-library.html

Reply:
I can try to write some mocha tests if you'd like: https://fable.io/docs/your-fable-project/author-a-fable-library.html#Testing

Reply:
Very cool. We'll see about adding those tests later on. 

>we are just adding the source code to the nuget package

Which NuGet Package?

Reply:
Hedgehog itself.

I will try to explain it step by step:
- Hedgehog build works normally just as before
- When the Hedgehog nuget package is created all of the source code of Hedgehog is added to the nuget package in fable subdirectory
- If somebody wants to use Hedgehog in a Fable project it adds Hedgehog nuget package to project dependencies
- Fable compiler compiles the project, finds Hedgehog nuget package and compiles the sources from the nuget package

For example if you clone my [example](https://github.com/ThisFunctionalTom/Fable.Hedgehog.Example) and type `npm test` the fable compiler will find the sources in the Hedgehog nuget package extract them into .fable directory and compile them to javascript.

Reply:
>Fable compiler compiles the project, finds Hedgehog nuget package and compiles the sources from the nuget package

Very nice :+1:  — Is it possible to test that this is going to succeed prior releasing a new Hedgehog version? e.g. as part of `dotnet test` or even `paket pack`

Reply:
I created a nuget package locally on my computer and used it in the example above, so if you clone this repo and try `npm test` it will compile the test project with fable and run the tests in it using nodejs.

Maybe we can put somthing simmilar in Hedgehog buildscript before publishing nuget package. 

For example the buildscrip could:
1. create nuget package locally
2. compile using fable compiler
3. run some mocha tests
4. publish nuget package if everything is OK.

Reply:
>I created a nuget package locally on my computer and used it in the example above

That's the repo, right? https://github.com/ThisFunctionalTom/Fable.Hedgehog.Example

Reply:
Yes. In it is hedgehog nuget package in directory nuget-packages.

Reply:
If you open this file [Hedgehog.0.8.3.1.nupkg](https://github.com/ThisFunctionalTom/Fable.Hedgehog.Example/blob/master/nuget-packages/Hedgehog.0.8.3.1.nupkg) with a zip tool you will see a fable directory with Hedgehog sources and project file in it. That is what fable compiler needs.

Reply:
I made two pull requests to Fable compiler which resolves our issues with the `using` function and with `StringBuilder.ToString(firstIndex, length)`. With Fable compiler 2.8 (which was release today) we do not need any extension anymore.

Reply:
This is awesome! Great work!

Reply:
I reverted my changes in Hedgehog.fsproj file and some spaces so now the pull request is clean again.
The only thing changed now are a few  `#if !FABLE_COMPILER` and the inclusion of source files in paket.template.